packages/gatsby-theme-docs/src/layouts/internals/layout-header.js

```diff
@@ -140,85 +143,110 @@ const SearchContainer = styled.div`
   }
 `;
 
-const LayoutHeader = (props) => (
-  <Container>
-    <Content>
-      <Inline alignItems="center">
-        <LogoContainer>
-          {/* Injected by React portal */}
-          <div
-            id="sidebar-menu-toggle"
-            css={css`
-              display: flex;
-              @media screen and (${designSystem.dimensions.viewports.laptop}) {
-                display: none;
-              }
-            `}
-          />
-          <LogoButton />
-        </LogoContainer>
-        <DocumentationSwitcherContainer isActive={props.isTopMenuOpen}>
-          <SwitcherButton
-            role="button"
-            aria-label={
-              props.isTopMenuOpen ? 'Close Top Menu' : 'Open Top Menu'
-            }
+const LayoutHeader = (props) => {
+  const [
+    mouseIsOverTopMenuButton,
+    setMouseIsOverTopMenuButton,
+  ] = React.useState(false);
+
+  return (
+    <Container>
+      <Content>
+        <Inline alignItems="center">
+          <LogoContainer>
+            {/* Injected by React portal */}
+            <div
+              id="sidebar-menu-toggle"
+              css={css`
+                display: flex;
+                @media screen and (${designSystem.dimensions.viewports
+                    .laptop}) {
+                  display: none;
+                }
+              `}
+            />
+            <LogoButton />
+          </LogoContainer>
+          <DocumentationSwitcherContainer
             isActive={props.isTopMenuOpen}
-            onClick={props.toggleTopMenu}
+            mouseIsOverTopMenuButton={mouseIsOverTopMenuButton}
           >
-            <SpacingsInline alignItems="center">
-              <span>{props.siteTitle}</span>
-              {props.isTopMenuOpen ? (
-                <AngleUpIcon size="medium" color="info" />
-              ) : (
-                <AngleDownIcon size="medium" />
-              )}
-            </SpacingsInline>
-          </SwitcherButton>
-          {props.isTopMenuOpen ? (
-            <Overlay
-              top={designSystem.dimensions.heights.header}
-              onClick={props.closeTopMenu}
+            <SwitcherButton
+              role="button"
+              aria-label={
+                props.isTopMenuOpen ? 'Close Top Menu' : 'Open Top Menu'
+              }
+              isActive={props.isTopMenuOpen}
+              onClick={props.toggleTopMenu}
+              onMouseOver={handleOnMouseOverTopMenu}
+              onMouseLeave={handleOnMouseLeaveTopMenu}
```

Comment:
Why do we need mouse events? We should be able to do this with CSS only.

Reply:
I don't have a solution for css only. Do you have any suggestion?

Reply:
I can take a look later but I would prefer to not use event listeners if not necessary.

Reply:
I pushed a commit with the changes: https://github.com/commercetools/commercetools-docs-kit/pull/581/commits/af399b88b5a17790f82360aec1bcb206258817f3

I also took the chance to improve the a11y of the button element (see gif, where I use tab navigation to focus on the element and then toggle the menu using the spacebar and the enter keys).

![2020-07-28 17 39 26](https://user-images.githubusercontent.com/1110551/88688438-9f85c680-d0f9-11ea-9dc0-8e62382e77b3.gif)